server/src/main/java/org/cloudfoundry/identity/uaa/provider/oauth/XOAuthAuthenticationManager.java

```diff
@@ -271,12 +271,12 @@ private String getTokenFromCode(XOAuthCodeToken codeToken, AbstractXOAuthIdentit
         }
 
         try {
-            if (config.isSkipSslValidation()) {
-                restTemplate.setRequestFactory(getNoValidatingClientHttpRequestFactory());
-            }
-            ResponseEntity<Map<String, String>> responseEntity = restTemplate.exchange(requestUri, HttpMethod.POST, requestEntity, new ParameterizedTypeReference<Map<String, String>>() {});
+            restTemplate.setRequestFactory(createRequestFactory(config.isSkipSslValidation()));
```

Comment:
before this change, when config.isSkipSslValidation == false, this code ended up using SimpleClientHttpRequestFactory  (via HttpAccessor class.)  Your change  replaces it with HttpComponentsClientHttpRequestFactory to allow use of system properties - that seems ok to me.